### PR TITLE
Use Caching for open files

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related: https://github.com/demisto/etc/issues/45825

## Description
Add caching to open files functions to improve validate run time.

### CODE REVIEW

**Comment by ChanochShayner** [comment]
@guykeller 
This cache mechanism downgrades the run time of:
Create ID Set - from 5.20 m to 3.20 m
Validate all files - from ~45 m to ~28 m

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -470,7 +470,10 @@ def get_last_remote_release_version():
     return ''
 
 
-def get_file(file_path, type_of_file):
+@lru_cache()
```
Comment (guykeller) [inline demisto_sdk/commands/common/tools.py:473]: 1. Have you tried changing the max size? What is our cache miss ratio?
2. Would love to discuss the cache invalidation strategy

**Comment by ChanochShayner** [comment]
Using the maxsize arg doesn't improve the runtime (even with maxsize=10000)
@guykeller FYI

**Comment by ilappe** [comment]
@ChanochShayner you can get the cache info (including miss, current size etc)
by printing the get_file.cache_info() (cache_info is a function added by the decorator)

**Review by guykeller — APPROVED** [review]

**Pull request merged**